modules/login/src/main/frontend/src/login/signUpForm.js

```diff
@@ -214,48 +215,42 @@ class SignUpForm extends React.Component {
         // HTTP codes, it'll only fail when network error
         // Therefore, you must handle the error code yourself.
         if (!response.ok) {
-          this.props.handleLogin && this.props.handleLogin(false);
+          handleLogin?.(false);
           response.json().then((data) => {
             let errMsg = data?.error?.message;
             errMsg = (errMsg ? errMsg : "Unknown Error");
-            this.setState({
-              errorOpen: true,
-              errorMsg: errMsg
-            });
-            this.form.setFieldError("username", errMsg);
+            setErrorOpen(true);
+            setErrorMsg(errMsg);
+            form.current.setFieldError("username", errMsg);
           })
           .catch(error => {
-            this.setState({
-              errorOpen: true,
-              errorMsg: "Unknown Error (JSON Parsing Failed)"
-            });
-            this.form.setFieldError("username", "Unknown Error (JSON Parsing Failed)");
+            setErrorOpen(true);
+            setErrorMsg("Unknown Error (JSON Parsing Failed)");
+            form.current.setFieldError("username", "Unknown Error (JSON Parsing Failed)");
           });
           throw Error(response.statusText);
         }
 
-        this.props.handleSuccess && this.props.handleSuccess();
-        this.props.loginOnSuccess && this.signIn(username, password);
+        handleSuccess && handleSuccess();
```

Comment:
Since we're making things shorter...
```suggestion
        handleSuccess?.();
```